scalableminds/webknossos
# Improve handling of movements in 3rd dimension

This PR removes the old "clamp to edge" behavior on initial third-dimension movements, since this had undesired side effects (such as #5233). As a replacement, the keyboard delay is adapted according to the current position. See code comments for more details.

### URL of deployed dev instance (used for testing):
https://fixvisiblemove.webknossos.xyz

### Steps to test:
- open a dataset (ideally with a real-world scale, such as 11,11,24)
- turn down the move value to something unbearably slow (e.g., 5 nm/s which I allowed temporarily for this PR)
- play around with the position field (which now supports floats in this PR <-- I will undo this)
  - set the z-value to x.99 and move forward with f (keep the key pressed)
  - set the z-value to x.00 and move forward with f (keep the key pressed)
  - the experienced keyboard delay for the first scenario should be higher than for the second. you can also check the console output to see the calculated delays.

### Issues:
- fixes #5233 
- hopefully softens #5653 

------
